shuffle/cli/src/deploy.rs

```diff
@@ -0,0 +1,218 @@
+
+use anyhow::{anyhow, Context, Result};
+use anyhow::{bail, format_err};
+use diem_config::config::NodeConfig;
+use diem_crypto::PrivateKey;
+use diem_sdk::{
+    client::BlockingClient,
+    transaction_builder::{Currency, TransactionFactory},
+    types::{
+        account_address::AccountAddress,
+        transaction::{Module, TransactionPayload},
+        LocalAccount,
+    },
+};
+use diem_temppath::TempPath;
+use diem_types::{
+    account_config, account_state::AccountState, account_state_blob::AccountStateBlob,
+    chain_id::ChainId, transaction::authenticator::AuthenticationKey,
+};
+use generate_key::load_key;
+use move_binary_format::{file_format::CompiledModule, normalized};
+use move_command_line_common::files::MOVE_EXTENSION;
+use shuffle_transaction_builder::framework::{
+    encode_create_parent_vasp_account_script_function, encode_mint_coin_script_function,
+};
+use std::{
+    convert::TryFrom,
+    fs, io,
+    io::Write,
+    path::{Path, PathBuf},
+    process::Command,
+};
+
+pub fn handle(project_dir: PathBuf, account_key_path: PathBuf) -> Result<()> {
+    let config_path = project_dir.join("nodeconfig/0").join("node.yaml");
+    let config = NodeConfig::load(&config_path)
+        .with_context(|| format!("Failed to load NodeConfig from file: {:?}", config_path))?;
+    let root_key_path = project_dir.join("nodeconfig").join("mint.key");
+    let root_account_key = load_key(root_key_path);
+    let new_account_key = load_key(account_key_path);
+    let json_rpc_url = format!("http://0.0.0.0:{}", config.json_rpc.address.port());
+    let factory = TransactionFactory::new(ChainId::test());
+    println!("Connecting to {}...", json_rpc_url);
+
+    let client = BlockingClient::new(json_rpc_url.clone());
+    let root_seq_num = client
+        .get_account(account_config::treasury_compliance_account_address())?
+        .into_inner()
+        .unwrap()
+        .sequence_number;
+    let mut root_account = LocalAccount::new(
+        account_config::treasury_compliance_account_address(),
+        root_account_key,
+        root_seq_num,
+    );
+    let mut new_account = LocalAccount::new(
+        AuthenticationKey::ed25519(&new_account_key.public_key()).derived_address(),
+        new_account_key,
+        0,
+    );
+    println!("======new account {}", new_account.address());
+
+    // Create a new account.
+    print!("Create a new ParentVASP account (we cannot create a regular account right now)...");
+    let create_new_account_txn = root_account.sign_with_transaction_builder(
+        TransactionFactory::new(ChainId::test()).payload(
+            encode_create_parent_vasp_account_script_function(
+                Currency::XUS.type_tag(),
+                0,
+                new_account.address(),
+                new_account.authentication_key().prefix().to_vec(),
+                vec![],
+                false,
+            ),
+        ),
+    );
+    send(&client, create_new_account_txn)?;
```

Comment:
here the client sends the account transaction to the local devnet right? what happens if the transaction is faulty?

Reply:
It would send transaction to whatever blockchain network you are connected to (determined by the jsonrpc url here). If faulty the transaction will fail